db/migrate/20210831105834_change_users_role_data_type_to_integer.rb

```diff
@@ -0,0 +1,9 @@
+class ChangeUsersRoleDataTypeToInteger < ActiveRecord::Migration[6.1]
+  def up
+    change_column(:users, :role, :integer, using: "role::integer")
```

Comment:
@jandudulski I wanted to test user's role, but it was string, not integer. So I was looking for a solution to change it, because typical "change column" didn't work. I had to ask senior dev to help me with that

Reply:
After long digging I found good explanation https://stackoverflow.com/questions/57148595/in-rails-activerecord-migrations-using-keyword

It seems that `using` keyword is supported since [rails 4.2](https://github.com/rails/rails/commit/fbef981fdc7ba64678f7ae1fc82b2cd790469280) but it's probably nowhere documented :scream:

Reply:
BTW - does it work? I mean - I wonder how Postgres will cast value like `admin` to the integer?

Reply:
@jandudulski I didn't  cast `admin` or other default `string` into` integer.` Firstly I deleted default "role" value , so it was a `nil`, later I had a possibility to cast `string ` to the `integer`.

Reply:
If there are no values in the DB you probably don't need to cast anything ;)

Reply:
@jandudulski I've just updated PR and added migration with (updated) schema for that

> If there are no values in the DB you probably don't need to cast anything ;)